src/conn/conn_log.c

```diff
@@ -69,15 +69,23 @@ __logmgr_force_ckpt(WT_SESSION_IMPL *session, uint32_t lognum)
 		 */
 		WT_RET(tmp_session->iface.checkpoint(
 		    &tmp_session->iface, "force=1"));
-		WT_RET(WT_SESSION_CHECK_PANIC(tmp_session));
 		/*
 		 * Only sleep in the rare case that we had to come through
 		 * this loop more than once.
 		 */
-		if (yield++) {
-			WT_STAT_CONN_INCR(session, log_force_ckpt_sleep);
-			__wt_sleep(0, WT_THOUSAND);
+		if (yield++ != 0) {
```

Comment:
We have a fairly new `__wt_spin_backoff` that implements a shared version of this. I'd also be totally fine with not tracking the yield stat here - and just tracking sleeps.